zproject/prod_settings_template.py

```diff
@@ -185,6 +186,89 @@
 #
 #SOCIAL_AUTH_SUBDOMAIN = 'auth'
 
+########
+# SAML Authentication
+#
+# In order to set up SAML authentication you will need to configure the
+# settings below, as explained in
+# https://python-social-auth.readthedocs.io/en/latest/backends/saml.html#required-configuration
+# using information given to you by your Identity Provider.
+# In the sample code below, example certificates are inserted for illustration,
+# but they have been modified to be invalid and you should replace them with
+# appropriate certificates of your own, as explained in the document linked above.
+#
+# On the IdP's side, you will have to specify the SSO URL and the Entity ID.
+# The Entity ID is just the value of the SOCIAL_AUTH_SAML_SP_ENTITY_ID setting below.
+# The SSO URL is at the /complete/saml/ endpoint, so the full address should be something like
+# https://yourzulipdomain.example.com/complete/saml/
+#
+# In the SOCIAL_AUTH_SAML_ENABLED_IDPS setting, an important part is the attribute mapping.
+# It should be configured in accordance with how the user attributes are named when sent by
+# your IdP. These names are often configurable in your IdP's interface, when setting up SAML
+# authentication (e.g. with Okta and G Suite providers, you explicitly specify them).
+# For instance, in the example configuration below:
+#         "attr_user_permanent_id": "email",
+#         "attr_first_name": "first_name",
+#         "attr_last_name": "last_name",
+#         "attr_username": "email",
+#         "attr_email": "email",
+# it is assumed the IdP specifies the user's email in the "email" atttribute,
+# the first name in the "first_name" attribute and the last name
+# in the "last_name" attribute.
+# You should generally only adjust the right side of this mapping to the proper attribute
+# names, as provided by your IdP - and leave the left side as it is.
+
+# SOCIAL_AUTH_SAML_SP_ENTITY_ID = 'https://' + EXTERNAL_HOST
+
+# For the public certificate and private key settings below, don't edit them directly,
+# but instead:
+# 1. Put the public certificate in /etc/zulip/saml/zulip-cert.crt
+# 2. Put the private key in /etc/zulip/saml/zulip-private-key.key
+# 3. For security reasons, set the correct ownership and permissions for these files:
+#   chown zulip.zulip /etc/zulip/saml/zulip-cert.crt /etc/zulip/saml/zulip-private-key.key
+#   chmod 644 /etc/zulip/saml/zulip-cert.crt
+#   chmod 640 /etc/zulip/saml/zulip-private-key.key
+
+# SOCIAL_AUTH_SAML_SP_PUBLIC_CERT = ""
+# SOCIAL_AUTH_SAML_SP_PRIVATE_KEY = ""
```

Comment:
I think we don't want these commented-out settings here at all if the recommendation is to not enter anything here, as it's just going to be confusing.

Reply:
Agreed, using your suggested changes to prod_settings_template now.